File: app/src/main/java/com/example/mobileproject/DB/CommentDB.java
package com.example.mobileproject.DB;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.example.mobileproject.Adapter.CommentRecyclerAdapter;
import com.example.mobileproject.holder.CommentItemHolder;
import com.example.mobileproject.holder.HomeItemHolder;
import com.example.mobileproject.model.CommentItem;
import com.example.mobileproject.model.DetailItem;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;
import com.google.firebase.firestore.QueryDocumentSnapshot;

public class CommentDB {
    private static final CommentDB instance = new CommentDB();

    private CommentDB() {}

    public static CommentDB getInstance() {

        return instance;
    }


    public void CommentPost(){
        FirebaseFirestore db = FirebaseFirestore.getInstance();
        FirebaseUser mUser = FirebaseAuth.getInstance().getCurrentUser();

        db.collection("post")
                .whereEqualTo("uid", mUser.getUid())
                .get()
                .addOnCompleteListener(task -> {
                    if (task.isSuccessful()) {
                        for (QueryDocumentSnapshot document : task.getResult()) {
                        }
                    } else {
                    }
                });
    }



}

File: app/src/main/java/com/example/mobileproject/DB/CommentDB.kt
package com.example.mobileproject.DB

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore

class CommentDB private constructor() {


    fun CommentPost() {
        val db = FirebaseFirestore.getInstance()
        val mUser = FirebaseAuth.getInstance().currentUser

        db.collection("post")
                .whereEqualTo("uid", mUser!!.uid)
                .get()
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        for (document in task.result!!) {
                        }
                    } else {
                    }
                }
    }

    companion object {
        val instance = CommentDB()
    }


}
